packages/lit-renderer/src/lit-renderer.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright (c) 2016 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { nothing, render } from 'lit';
+import { AsyncDirective } from 'lit/async-directive.js';
+import { PartType } from 'lit/directive.js';
+
+// A symbol indicating that the directive hasn't been initialized.
+const VALUE_NOT_INITIALIZED = Symbol('valueNotInitialized');
+
+export class LitRendererDirective extends AsyncDirective {
+  /** @protected */
+  previousValue = VALUE_NOT_INITIALIZED;
+
+  constructor(part) {
+    super(part);
+    if (part.type !== PartType.ELEMENT) {
+      throw new Error(`The \`${this.name}\` directive must be bound to an element.`);
+    }
+  }
+
+  /** @override */
+  update(part, [renderer, value]) {
+    if (!this.hasChanged(value)) {
+      return nothing;
+    }
+
+    this.host = part.options && part.options.host;
+    this.element = part.element;
+    this.renderer = renderer;
+
+    const firstRender = this.previousValue === VALUE_NOT_INITIALIZED;
+    if (firstRender) {
+      this.addRenderer();
+    } else {
+      this.runRenderer();
+    }
+
+    // Copy the value if it is an array in order to keep it
+    // from possible outside mutations.
+    this.previousValue = Array.isArray(value) ? [...value] : value;
+
+    return nothing;
+  }
+
+  /** @override */
+  reconnected() {
+    this.addRenderer();
+  }
+
+  /** @override */
+  disconnected() {
+    this.removeRenderer();
+  }
+
+  /** @abstract */
+  get name() {
+    throw new Error('The `name` getter must be implemented.');
```

Comment:
Consider using native [`Function.name`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name) e.g. like this:

![class-name](https://user-images.githubusercontent.com/10589913/165760534-bba53481-9362-4b5c-9664-721c487007f3.png)

Reply:
I was thinking of that too, but I was afraid that there is a chance that the name can be mangled by a compressor. 

However, now I think that it is probably most relevant to the production build which is not a focus of our attention in this PR, so I'll update it as you suggested.

Reply:
Right, I forgot that we currently don't export these directive classes. 
Maybe we could do that in case someone wants to extend them?